apps/bgu/CMakeLists.txt

```diff
@@ -12,8 +12,7 @@ set(CMAKE_CXX_EXTENSIONS NO)
 find_package(Halide REQUIRED)
 
 # Generator
-add_executable(bgu.generator bgu_generator.cpp)
-target_link_libraries(bgu.generator PRIVATE Halide::Generator Halide::Tools)
+add_halide_generator(bgu.generator SOURCES bgu_generator.cpp)
 
 # Filters
 add_halide_library(bgu FROM bgu.generator)
```

Comment:
When using `add_halide_generator`, you need to use the namespaced name to get correct cross-compiling behavior. This would need to be:

```
add_halide_library(bgu FROM bgu::halide_generators::bgu.generator)
```

This needs to be adjusted everywhere.

Reply:
I hear you, and yet, it seemed to work properly for me locally.

Reply:
That is because you aren't cross-compiling. When you try to cross-compile these, it will fail.

Reply:
The plain-named target exists in the host build to allow for build customization... the pattern is:

```
add_halide_generator(gen SOURCES ...)
if (TARGET gen)
  # set gen's properties, etc.
endif ()

# project::halide_generators::gen either comes from a host build import or is an alias to plain "gen"
add_halide_library(lib FROM project::halide_generators::gen)
```

Reply:
(1) Fair enough, but I don't understand why the previous code wouldn't also fail in those cases. 
(2) Can we / should we try to make it so that it will fail reliably in all cases?

Reply:
(1) the old code (as in, pre-PR) _would_ fail to cross-compile, yes.

(2) I am super open to ideas. I think we're pretty free to change `add_halide_generator` since it's quite new.

Reply:
Gotcha.

Couldn't we have `add_halide_library()` check the `FROM` and if it's not present, look in `PACKAGE::halide_generators::FROM`?

Reply:
> Couldn't we have `add_halide_library()` check the `FROM` and if it's not present, look in `PACKAGE::halide_generators::FROM`?

We absolutely could do that.